modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java

```diff
@@ -0,0 +1,362 @@
+package io.swagger.codegen.languages;
+
+import java.io.File;
+import java.util.*;
+
+import io.swagger.codegen.*;
+import io.swagger.models.Model;
+import io.swagger.models.Operation;
+import io.swagger.models.Response;
+import io.swagger.models.Swagger;
+import io.swagger.models.properties.*;
+
+public class AdaCodegen extends AbstractAdaCodegen implements CodegenConfig {
+    protected String packageName = "swagger";
+    protected String projectName = "Swagger";
+    protected List<Map<String, Object>> orderedModels;
+    protected Map<String, List<String>> modelDepends;
+
+    public AdaCodegen() {
+        super();
+
+        modelNameSuffix = "_Type";
+        orderedModels = new ArrayList<Map<String, Object>>();
+        modelDepends = new HashMap<String, List<String>>();
+        embeddedTemplateDir = templateDir = "Ada";
+
+        // CLI options
+        addOption(CodegenConstants.PROJECT_NAME, "GNAT project name",
+                  this.projectName);
+        addOption(CodegenConstants.PACKAGE_NAME, "Ada package name (convention: name.space.model).",
+                  this.modelPackage);
+        addOption(CodegenConstants.MODEL_PACKAGE, "Ada package for models (convention: name.space.model).",
+                  this.modelPackage);
+        addOption(CodegenConstants.API_PACKAGE, "Ada package for apis (convention: name.space.api).",
+                  this.apiPackage);
+
+        languageSpecificPrimitives = new HashSet<String>(
+                Arrays.asList("integer", "boolean", "Integer", "Character", "Boolean", "long", "float", "double", "int32_t", "int64_t"));
+
+        typeMapping = new HashMap<String, String>();
+        typeMapping.put("date", "Swagger.Date");
+        typeMapping.put("DateTime", "Swagger.Datetime");
+        typeMapping.put("string", "Swagger.UString");
+        typeMapping.put("integer", "Integer");
+        typeMapping.put("long", "Swagger.Long");
+        typeMapping.put("boolean", "Boolean");
+        typeMapping.put("array", "Swagger.Vector");
+        typeMapping.put("map", "Swagger.Map");
+        typeMapping.put("object", "Swagger.Object");
+        typeMapping.put("number", "Swagger.Number");
+        typeMapping.put("UUID", "Swagger.UString");
+        typeMapping.put("file", "Swagger.Http_Content_Type");
+        typeMapping.put("binary", "Swagger.Binary");
+
+        super.importMapping = new HashMap<String, String>();
+    }
+
+    @Override
+    public CodegenType getTag() {
+        return CodegenType.CONFIG;
+    }
+
+    @Override
+    public String getName() {
+        return "Ada";
```

Comment:
I'll rename it to "ada" as the convention is [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)

Ref: https://github.com/swagger-api/swagger-codegen/wiki/How-to-add-a-generator-for-a-new-language-or-framework#checklist-a-list-of-items-we-look-for-in-the-new-generator

Reply:
Ok. For your info, Ada is the name of a person: Ada Lovelace, which is why it has a capital letter.
I'll do the changes this evening. Thanks for taking time to review my pull request.

Looking at other generators, CsharpDotNet2 is not compliant then as it uses:

CsharpDotNet2ClientCodegen.java-        return "CsharpDotNet2";

Reply:
> Ok. For your info, Ada is the name of a person: Ada Lovelace, which is why it has a capital letter.

Yup, I read that from https://en.wikipedia.org/wiki/Ada_(programming_language)

> CsharpDotNet2

Yes, I planned to change that (as reminded by this PR). There are only 2 that needs to be updated to conform to kebab-case.

Reply:
FYI. PRs for renaming the other 2 generators:
https://github.com/swagger-api/swagger-codegen/pull/6612
https://github.com/swagger-api/swagger-codegen/pull/6611